src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,93 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 1 << 4;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int capacity;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = getHash(key);
+        Node<K,V> newNode = new Node<>(hash, key, value, null);
+        if (table[hash] == null) {
+            table[hash] = newNode;
+            size++;
+            return;
+        }
+        Node<K,V> currentNode = table[hash];
+        while (true) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                break;
+            }
+            currentNode = currentNode.next;
+        }
```

Comment:
Could you refactor it so that not use the break operator and replace while(true) with another condition?